src/wavm/WAVMWasmModule.cpp

```diff
@@ -989,7 +994,11 @@ U32 WAVMWasmModule::mmapFile(U32 fd, U32 length)
 
 U32 WAVMWasmModule::allocateThreadStack()
 {
-    return mmapMemory(THREAD_STACK_SIZE);
+    createMemoryGuardRegion();
+    U32 wasmPtr = mmapMemory(THREAD_STACK_SIZE);
+    createMemoryGuardRegion();
```

Comment:
Stacks that we provision ourselves should always be sandwiched between guard regions to catch over/ underflows